### PR TITLE
test: added test for src/graphql/types/Mutation/createFundCampaignPledge

### PR DESCRIPTION
**Pull request by MonishPuttu**
Added integration tests for src/graphql/types/Mutation/createFundCampaignPledge
 
Fixes: #3860

15 Test Cases Written:

Authentication:

- Unauthenticated user error
- Deleted user edge case 

Input Validation:

- Invalid campaign/pledger UUID format
- Negative amount validation

Resource Validation:

- Campaign/pledger not found errors
- Both resources not found

Business Logic:

- Duplicate pledge prevention
- Campaign ended/not started validation

Success Scenarios:

- Complete pledge creation
- Minimal pledge creation 
- Multiple pledges by different users
- Various valid amount values

Coverage Achieved: 

1. Statement: 96.23%
2. Branch: 96.55%
3. Function: 100%

Uncovered Lines: 233-241 (defensive database error check)

Lines 233-241 handle a rare edge case where database insert succeeds without exception but returns empty results. This defensive check cannot be reliably tested via integration tests without mocking the database layer.

"should istanbul ignore comment be added for this"

Files Modified:
test/graphql/types/Mutation/createFundCampaignPledge.test.ts - Complete test suite (15 tests)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive integration tests for fund campaign pledges covering positive flows, extensive validation (amounts, UUIDs), authentication/authorization scenarios, error cases (non‑existent or already‑pledged campaigns, timing constraints), multiple pledgers, and teardown/cleanup to ensure test isolation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: test/graphql/types/Mutation/createFundCampaignPledge.test.ts [hunk 1]
```diff
@@ -0,0 +1,904 @@
...
+			);
+
+			expect(result.errors).toBeUndefined();
+			expect(result.data?.createFundCampaignPledge?.amount).toBe(amount);
+
+			if (result.data?.createFundCampaignPledge?.id) {
+				createdPledgeIds.push(result.data.createFundCampaignPledge.id);
+			}
+		}
+	});
```
[the remark this answers is not in the document]
Comment (MonishPuttu) [reply test/graphql/types/Mutation/createFundCampaignPledge.test.ts:941]: working on it

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all resolved items and include the following points:
1. Validate whether any updated test files:
    1. Cover all use cases
    1. Have valid tests
    1. Provide 100% test code coverage
    1. Require any additional tests
1. Identify:
    1. Lines of code that are missing tests.
    1. Additional tests that may be required
    1. Any unnecessary files that have been submitted and ask that they be excluded.
1. Validate that the PR:
    1. Meets the requirements of the linked issue.
    1. Complies with all previously suggested improvements and comments.

If the submitted code requires any changes, even trivial ones, then make coderabbitai[bot] give the review a CHANGES_REQUESTED status. If no changes are required then give the PR an APPROVED status.

**Comment by palisadoes** [comment]
You may now use the repo's issue auto-assignment feature outlined here: ISSUE_GUIDELINES.md
